app/src/main/java/net/gsantner/markor/format/markdown/MarkdownTextConverter.java

```diff
@@ -323,4 +377,62 @@ public boolean isFileOutOfThisFormat(String filepath) {
         filepath = filepath.replace(JavaPasswordbasedCryption.DEFAULT_ENCRYPTION_EXTENSION, "");
         return (MarkdownTextConverter.PATTERN_HAS_FILE_EXTENSION_FOR_THIS_FORMAT.matcher(filepath).matches() && !filepath.toLowerCase().endsWith(".txt")) || filepath.toLowerCase().endsWith(".md.txt");
     }
+
+    private Map<String, List<String>> extractYamlAttributes(final String markup) {
+        Parser yamlFrontmatterParser = Parser.builder().extensions(Collections.singleton(YamlFrontMatterExtension.create())).build();
+        AbstractYamlFrontMatterVisitor visitor = new AbstractYamlFrontMatterVisitor();
+        Node document = yamlFrontmatterParser.parse(markup);
+        visitor.visit(document);
+        return visitor.getData();
+    }
+
+    private String replaceTokens(final String markup, final String scopes, boolean wrapInDiv) {
+        String markupReplaced = markup;
+        String attrVal_S = "";
+        String attrVal_E = "";
+
+        for (Map.Entry<String, List<String>> entry : yamlAttributeMap.entrySet()) {
+            String attrName = entry.getKey();
+            List<String> attrValue = entry.getValue();
+            List<String> attrValueOut = new ArrayList<>();
+
+            if (attrName.equals("tags") && attrValue.size() == 1) {
+                // It's not a real tag list, but rather a string of comma-separated strings
+                attrValue = Arrays.asList(attrValue.get(0).split(",\\s*"));
+            }
+
+            if (wrapInDiv) {
+                attrVal_S = HTML_TOKEN_ITEM_CONTAINER_S;
+                attrVal_E = HTML_TOKEN_ITEM_CONTAINER_E + "\n";
+            } else {
+                attrVal_S = "";
+                attrVal_E = "";
+            }
+            for (String aValue : attrValue) {
+                // Strip surrounding single or double quotes
+                aValue = aValue.replaceFirst("^(['\"])(.*)\\1", "$2");
+                aValue = TextUtils.htmlEncode(aValue);
+                aValue = aValue.replaceAll("`(.*?)`", "<code>$1</code>");
+                aValue = aValue.replaceAll("_(.*?)_", "<em>$1</em>");
+                aValue = aValue.replaceAll("\\*(.*?)\\*", "<b>$1</b>");
+                aValue = aValue.replaceAll("(?<!-)---(?!-)", "&mdash;");
+                aValue = aValue.replaceAll("(?<!-)--(?!-)", "&ndash;");
+                aValue = aValue.replaceAll("\\$(.*?)\\$", "<span class='katex'>$1</span>");
```

Comment:
I like it that it currently supports markdown. However I have to say - it's incompatible with everything else. I use frontmatter on various things, and it always just pastes the text without doing markdown converting inside that as well.

Text is often quite unpredictable and hard to know in advance all cases - but if anything, at Markor we've seen that even the most advanced regex fits all cases. What I'd keep is the things like html encoding and quotes. But I'd remove code, em, katex and b. If the user wants to have something bold, they can also use css for that.

![image](https://user-images.githubusercontent.com/6735650/154805897-bb52ce1b-9bed-4929-9a09-c7c916f5cf24.png)

Preferably I'd like to not introduce something that doesn't work on any other markdown software / converter.